protocol/message_persistence.go

```diff
@@ -2012,6 +2012,34 @@ func (db sqlitePersistence) GetEdits(messageID string, from string) ([]*EditMess
 	return messages, nil
 }
 
+func (db sqlitePersistence) GetUnviewedCounts(id string) (messageCounts uint, mentionCounts uint, err error) {
```

Comment:
This is going to be too slow, you remember that we discussed another method in our call?